File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string answer= get_string("text:\n");
    int n=strlen(answer);
    int letter=0;
    int word=1;
    int sentence=0;
    for(int i = 0; i < n; i++){
        if ((answer[i] != 0) && (((answer[i] < 123) && (answer[i] > 96)) || ((answer[i] <91) && (answer[i] > 64)))){
            letter++;
        }
        if ((answer[i]!=' ') && (answer[i+1]==' '))
        {
            word++;
        }
        if (answer[i] == '.' || answer[i] == '?' || answer[i] == '!')
        {
            sentence++;
        }
    }
    float L = (100 * letter) / word;
    float S = (100 * sentence) / word;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    float I = round(index);
    if (I >= 16)
    {
        printf("Grade 16+\n");
    }
    if (I < 1)
    {
        printf("Before Grade 1\n");
    }
    if ((I >= 1) && (I < 16))
    {
        printf("Grade %0.f\n", I);

    }

}

File: pset6/readability.py
from cs50 import get_string
s = get_string("text: \n").strip()

letter = 0
word = 0
sentence = 0

for i in range(len(s)):
    if s[i].isalpha():
        letter += 1
    if (i == 0 and s[i] != ' ') or (i != len(s) - 1 and s[i] == ' ' and s[i+1] != ' '):
        word += 1
    if s[i] == '.' or s[i] == '?' or s[i] == '!':
        sentence += 1

L = 100 * letter / word
S = 100 * sentence / word
I = round(0.0588 * L - 0.296 * S - 15.8)

if I >= 16:
    print("Grade 16+")
if I < 1:
    print("Before Grade 1")
if I >= 1 and I < 16:
    print(f"Grade {I}")